### ISSUE TITLE
SharedArrayBuffer does not work with Firefox on Android anymore on itch.io

### ISSUE THREAD

**Issue by Gwynlix** [godotengine/godot#86988]
### Tested versions

- Reproducible in Godot 4.2.1

### System information

Windows 10 - Godot 4.2.1

### Issue description

A few days ago my Godot game stopped working on Itch.IO when opened from a Firefox Android app. It shows the following screen:

![grafik](https://github.com/godotengine/godot/assets/14159695/2d9789ca-8a10-40c4-bc94-2009cba83434)

On any other device I tested (Firefox, Chrome, Edge on PC and Chrome on Android) it still works like a charm.

### Steps to reproduce

You can check it on https://gwynlix.itch.io/auto-dwarfs - all I did was doing a web HTML5 export, upload it on itch with the SharedArrayBuffer option enabled and start the game.

### Minimal reproduction project (MRP)

N/A - it's export related and the project itself is actually irrelevant.

**Comment by akien-mga** [comment]
I can confirm the issue on Firefox Beta on Android.

@leafo Did something change recently in itchio's SharedArrayBuffer support, or should we look at a potential Firefox regression / them maybe disabling the origin trial for `credentialless`?

**Comment by adamscott** [comment]
![Screenshot_20240109-081712.png](https://github.com/godotengine/godot/assets/270928/22549302-e969-4b06-b947-f525ac15cbf2)

Works on my Pixel 7 Pro Firefox Nightly.

**Comment by akien-mga** [comment]
Corresponding itch.io bug report: https://itch.io/t/3405599/godot-4-html5-on-firefox-android-currently-broken-4-jan-2024

> Works on my Pixel 7 Pro Firefox Nightly (Firefox 123). So if it's a regression, it seems fixed on the main Firefox branch!

I can confirm it works on Nightly, but that doesn't necessarily mean it's fixed. Nightly uses a different default configuration, which doesn't automatically make its way to Beta/Stable branches.

Here it seems indeed that on Android, Firefox Nightly enables `browser.tabs.remote.coep.credentialless` by default, while it's off on Firefox Beta (and likely Release). Setting it to `true` on Beta makes the test project work again.

There's also a `dom.origin-trials.coep-credentialless.state` which is `0` for both Beta and Nightly. Turning it to `1` on Beta seems to also make the demo work.

I doubt Beta/Release had `coep-credentialless` enabled by default for all websites a few days ago (the first setting), but maybe the value for the `origin-trials` state change, whatever it implies.

**Comment by MrMinimal** [comment]
Still present, with Godot 4.2.1-stable

**Comment by akien-mga** [comment]
Based on this thread, the issue seems indeed that the Firefox origin trial for itch.io expired: https://twitter.com/RancidBacon/status/1777742231979647232

@leafo is looking into it, but made a change in the meantime so that games start in a popup on Firefox, which allows the proper headers to be set for SharedArrayBuffer: https://twitter.com/moonscript/status/1777805596097732816

**Comment by Rult** [comment]
Godot 4.3 seems to resolve this issue by making games single threaded by default, so you can turn off "SharedArrayBuffer support" on your itch.io game page settings and it will work without opening a new window for a game.

**Comment by synalice** [comment]
Mobile Firefox 133.0.3. Could not reproduce the bug. Linked game successfully launches in a separate `https://itch.io/embed-upload/...` tab.

@Gwynlix, can we close this?

**Comment by akien-mga** [comment]
Yes itch.io implemented the behavior of running SharedArrayBuffer games in a separate tab on Firefox to solve this issue.
